### PR TITLE
[enhancement](Nereids)make nereids work

### PR DESCRIPTION
**Pull request by morrySnow**
# Proposed changes

Nereids could execute query: `select a from t;`

**enhancement**
- add a queriable interface for QueryStmt and LogicalPlanAdapter Temporarily
- refactor GroupId, GroupId extends doris.common.id now
- GroupId is generated by it's memo now, not global yet
- add varchar type
- Nereids enabled only when vectorized engine enabled

**fix**
- set output and column label to logicalPlanAdapter
- set output expression on root fragment
- set select partition and select index id to OlapScanNode
- BatchRulesJob add rule type mismatch
- add all implementation rules to rule set
- SlotReference get catalog column no longer returns null values
- bind star correctly
- implement `isNullable` in expressions

**known issue**
- could not do expression mapping(e.g. a + 1) on project node(wait intermediate tuple interface and project ability in ExecNode in be)
- aggregate do not work
- sort do not work
- filter do not work
- join do not work

## Checklist(Required)

1. Does it affect the original behavior: No
2. Has unit tests been added: No
3. Has document been added or modified: NoNeed
4. Does it need to update dependencies: No
5. Are there any changes that cannot be rolled back: No

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 1]
```diff
@@ -169,6 +172,12 @@ public PlanFragment visitPhysicalOlapScan(
         TableRef ref = new TableRef(tableName, null, null);
         BaseTableRef tableRef = new BaseTableRef(ref, olapTable, tableName);
         tupleDescriptor.setRef(tableRef);
+        olapScanNode.setSelectedPartitionIds(physicalOlapScan.getSelectedPartitionId());
+        try {
+            olapScanNode.updateScanRangeInfoByNewMVSelector(physicalOlapScan.getSelectedIndexId(), false, "");
+        } catch (Exception e) {
+            LOG.warn("set failed", e);
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:179]: what happens when an exception is caught? compute wrong result?

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:179]: change to throw exception

File: fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java [hunk 2]
```diff
@@ -358,7 +358,12 @@ public void init(Analyzer analyzer) throws UserException {
     }
 
     public void init() throws UserException {
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java:360]: Add a comment to explain `This function is only used for nereids `

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java:360]: 👌

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**